lib/stub_ui/lib/src/ui/text.dart

```diff
@@ -1117,6 +1296,13 @@ abstract class Paragraph {
   /// Coordinates of the [TextBox] are relative to the upper-left corner of the paragraph,
   /// where positive y values indicate down.
   List<TextBox> getBoxesForPlaceholders();
+
+  /// Returns the full list of [LineMetrics] that describe in detail the various
+  /// metrics of each laid out line.
+  ///
+  /// This can potentially return a large amount of data, so it is not recommended
+  /// to repeatedly call this. Instead, cache the results.
+  List<LineMetrics> getLineMetrics();
```

Comment:
i would call this `computeLineMetrics` to make it sound expensive. Things called `getFoo` should really be getters, and those look cheap.